contracts/pcv/morpho/MorphoCompoundPCVDeposit.sol

```diff
@@ -0,0 +1,385 @@
+    /// then writes the current amount of PCV tracked in this contract
+    /// to lastRecordedBalance
+    /// @return the amount deposited after adding accrued interest or realizing losses
+    function accrue() external nonReentrant whenNotPaused returns (uint256) {
+        int256 startingRecordedBalance = lastRecordedBalance.toInt256();
+
+        _recordPNL(); /// update deposit amount and fire harvest event
+
+        int256 endingRecordedBalance = lastRecordedBalance.toInt256();
+
+        if (pcvOracle != address(0)) {
+            /// if any amount of PCV is withdrawn and no gains, delta is negative
+            IPCVOracle(pcvOracle).updateLiquidBalance(
+                endingRecordedBalance - startingRecordedBalance
+            );
+        }
+
+        return lastRecordedBalance; /// return updated pcv amount
+    }
+
+    /// ------------------------------------------
+    /// ------------ Permissioned API ------------
+    /// ------------------------------------------
+
+    /// @notice withdraw tokens from the PCV allocation
+    /// non-reentrant as state changes and external calls are made
+    /// @param to the address PCV will be sent to
+    /// @param amount of tokens withdrawn
+    function withdraw(address to, uint256 amount)
+        external
+        onlyPCVController
+        nonReentrant
+    {
+        int256 startingRecordedBalance = lastRecordedBalance.toInt256();
+
+        _withdraw(to, amount, true);
+
+        int256 endingRecordedBalance = lastRecordedBalance.toInt256();
+        if (pcvOracle != address(0)) {
+            /// if any amount of PCV is withdrawn and no gains, delta is negative
+            IPCVOracle(pcvOracle).updateLiquidBalance(
+                endingRecordedBalance - startingRecordedBalance
+            );
+        }
+    }
+
+    /// @notice withdraw all tokens from Morpho
+    /// non-reentrant as state changes and external calls are made
+    /// @param to the address PCV will be sent to
+    function withdrawAll(address to) external onlyPCVController nonReentrant {
+        int256 startingRecordedBalance = lastRecordedBalance.toInt256();
+
+        /// compute profit from interest accrued and emit an event
+        _recordPNL();
+
+        /// withdraw last recorded amount as this was updated in record pnl
+        _withdraw(to, lastRecordedBalance, false);
+
+        int256 endingRecordedBalance = lastRecordedBalance.toInt256();
+
+        if (pcvOracle != address(0)) {
+            /// all PCV withdrawn, send call in with amount withdrawn negative if any amount is withdrawn
+            IPCVOracle(pcvOracle).updateLiquidBalance(
+                endingRecordedBalance - startingRecordedBalance
+            );
+        }
+    }
+
+    /// @notice set the pcv oracle address
+    /// @param _pcvOracle new pcv oracle to reference
+    function setPCVOracle(address _pcvOracle) external onlyGovernor {
+        address oldOracle = pcvOracle;
+        pcvOracle = _pcvOracle;
```

Comment:
This should then call into the oracle with the delta being the current balance so things are recorded in their current state.